dis_snek/models/command.py

```diff
@@ -98,6 +108,47 @@ async def __call__(self, context, *args, **kwargs):
             if self.max_concurrency is not MISSING:
                 await self.max_concurrency.release(context)
 
+    async def call_callback(self, callback: Callable, context: "Context"):
+        callback = functools.partial(callback, context)  # first param must be ctx
+        parameters = get_parameters(callback)
+        args = []
+        kwargs = {}
+        if len(parameters) == 0:
+            # if no params, user only wants context
+            return await callback()
+
+        c_args = copy.copy(context.args)
+        for param in parameters.values():
+            if config := getattr(param.annotation, "_annotation_dat", None):
+                # if user has used an snek-annotation, run the annotation, and pass the result to the user
+                local = {"context": context, "scale": self.scale}
+                ano_args = [local[c] for c in config["args"]]
+                if param.kind != param.POSITIONAL_ONLY:
+                    kwargs[param.name] = param.annotation(*ano_args)
```

Comment:
No async anno function support?

Reply:
Calling a coroutine is slower than calling a regular function, baring in mind this code would be used by application commands, it seems like a *very* bad idea to allow for coroutines, which could potentitally be blocked by the event loop for an indeterminate amount of time, causing interactions to fail